addons/web/static/src/js/views/graph/graph_owl_renderer.js

```diff
@@ -0,0 +1,927 @@
+                }
+            }
+            return index;
+        }
+
+        _getDatasetLabel(dataPt) {
+            if (_.contains(['bar', 'line'], this.props.mode)) {
+                // ([origin] + second to last groupBys) or measure
+                var datasetLabel = dataPt.labels.slice(1).join("/");
+                if (this.props.origins.length > 1) {
+                    datasetLabel = this.props.origins[dataPt.originIndex] +
+                        (datasetLabel ? ('/' + datasetLabel) : '');
+                }
+                datasetLabel = datasetLabel || this.fields[this.props.measure].string;
+                return datasetLabel;
+            }
+            return this.props.origins[dataPt.originIndex];
+        }
+
+        _getDatasetDataLength(originIndex, defaultLength) {
+            if (_.contains(['bar', 'line'], this.props.mode) && this.props.comparisonFieldIndex === 0) {
+                return this.dateClasses.dateSets[originIndex].length;
+            }
+            return defaultLength;
+        }
+
+        _prepareData(dataPoints) {
+            var self = this;
+
+            var labels = dataPoints.reduce(
+                function (acc, dataPt) {
+                    var label = self._getLabel(dataPt);
+                    var index = self._indexOf(acc, label);
+                    if (index === -1) {
+                        acc.push(label);
+                    }
+                    return acc;
+                },
+                []
+            );
+
+            var newDataset = function (datasetLabel, originIndex) {
+                var data = new Array(self._getDatasetDataLength(originIndex, labels.length)).fill(0);
+                return {
+                    label: datasetLabel,
+                    data: data,
+                    originIndex: originIndex,
+                };
+            };
+
+            // dataPoints --> datasets
+            var datasets = _.values(dataPoints.reduce(
+                function (acc, dataPt) {
+                    var datasetLabel = self._getDatasetLabel(dataPt);
+                    if (!(datasetLabel in acc)) {
+                        acc[datasetLabel] = newDataset(datasetLabel, dataPt.originIndex);
+                    }
+                    var label = self._getLabel(dataPt);
+                    var labelIndex = self._indexOf(labels, label);
+                    acc[datasetLabel].data[labelIndex] = dataPt.value;
+                    return acc;
+                },
+                {}
+            ));
+
+            // sort by origin
+            datasets = datasets.sort(function (dataset1, dataset2) {
+                return dataset1.originIndex - dataset2.originIndex;
+            });
+
+            return {
+                datasets: datasets,
+                labels: labels,
+            };
+        }
+
+        _getLabel(dataPt) {
+            var i = this.props.comparisonFieldIndex;
+            if (_.contains(['bar', 'line'], this.props.mode)) {
```

Comment:
_.contains can be replace with native method includes on Array (see MDN for js doc)

Reply:
In general, we try to not use underscore.js and jQuery